setup.py

```diff
@@ -419,100 +13,53 @@ def find_tbb(tbb_root=None, tbb_include=None, tbb_link=None):
-    parallel_err = "file not recognized: file truncated"
-    tbb_err = "'tbb/tbb.h' file not found"
-
-    err_out = tfile.read().decode('utf-8')
-    sys.stderr.write(err_out)
-    if tbb_err in err_out:
-        sys.stderr.write("\n\033[1mUnable to find tbb. If you have TBB on "
-                         "your system, try specifying the location using the "
-                         "--TBB-ROOT or the --TBB-INCLUDE/--TBB-LINK "
-                         "arguments to setup.py.\033[0m\n")
-    elif parallel_err in err_out and args.nthreads > 1:
-        sys.stderr.write("\n\033[1mYou attempted parallel compilation on a "
-                         "Python version where this leads to a race "
-                         "in distutils. Please recompile without the -j "
-                         "option and try again.\033[0m\n")
-    else:
-        traceback.print_exc(limit=1)
-except: # noqa
-    sys.stderr.write(tfile.read().decode('utf-8'))
-    traceback.print_exc(limit=1)
-else:
-    if args.print_warnings:
-        sys.stderr.write("Printing warnings:\n")
-        sys.stderr.write(tfile.read().decode('utf-8'))
-    else:
-        out = tfile.read().decode('utf-8')
-        if out:
-            sys.stdout.write("Some warnings were emitted during compilations. "
-                             "Call setup.py with the {} argument "
-                             "to see these warnings.\n".format(warnings_str))
+setup(
+    name='freud-analysis',
+    version=version,
+    packages=['freud'],
+    description=desc,
+    long_description=readme,
+    long_description_content_type='text/x-rst',
+    keywords=('simulation analysis molecular dynamics soft matter '
+              'particle system computational physics'),
+    classifiers=[
+        "Development Status :: 5 - Production/Stable",
+        "Intended Audience :: Science/Research",
+        "License :: OSI Approved :: BSD License",
+        "Topic :: Scientific/Engineering :: Chemistry",
+        "Topic :: Scientific/Engineering :: Physics",
+        "Programming Language :: C++",
+        "Programming Language :: Cython",
+        "Programming Language :: Python :: 3.6",
+        "Programming Language :: Python :: 3.7",
+        "Programming Language :: Python :: 3.8",
+    ],
+    zip_safe=False,
+    maintainer='freud Developers',
+    maintainer_email='freud-developers@umich.edu',
+    # See documentation credits for current and former lead developers
+    author='Vyas Ramasubramani et al.',
+    author_email='vramasub@umich.edu',
+    url='https://github.com/glotzerlab/freud',
+    download_url='https://pypi.org/project/freud-analysis/',
+    project_urls={
+        'Homepage': 'https://github.com/glotzerlab/freud',
+        'Documentation': 'https://freud.readthedocs.io/',
+        'Source Code': 'https://github.com/glotzerlab/freud',
+        'Issue Tracker': 'https://github.com/glotzerlab/freud/issues',
+    },
+    python_requires='>=3.6',
+    install_requires=[
```

Comment:
Does this need to list scikit-build and cmake? Not sure how these interact with pyproject.toml.

Reply:
No, the `install_requires` is to determine dependencies of the code, whereas pyproject.toml is for dependencies of the build process. The requirements in pyproject.toml indicate what needs to be installed to even run setup.py.